edx/cc2olx
# [BD-27] [TNL-7664] [BB-3166] Add a way to support third party video as xblocks

This PR adds a functionality to parse a csv file and add the given link as xblock from iframe

**JIRA tickets**: [TNL-7664](https://openedx.atlassian.net/browse/TNL-7664)

~~**Discussions**: Link to any public dicussions about this PR or the design/architecture. Otherwise omit this.~~

~~**Dependencies**: None~~

~~**Screenshots**: Always include screenshots if there is any change to the UI.~~

~~**Sandbox URL**: TBD - sandbox is being provisioned.~~

~~**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.~~

**Testing instructions**:

We need a zip file to the corresponding course, we are going to use the test file we have:

Make sure to change to this branch and install the **requirement** and the tool in a **virtualenv**.

1.  `cd cc2olx`
1.   `cd tests/fixtures_data/imscc_file`
1.   `zip -r /path/to/<filename>.imscc *`
1.  `cc2olx -i /path/to/<filename>.imscc  -f tests/fixtures_data/link_map.csv`, makes sure to remove all the value of "Edx Id" column from `tests/fixtures_data/link_map.csv` while keeping the header intact
1. In the CWD there will be output folder created, there will be a "tar.gz", use that tar.gz file to import the course

In the course go to video tab you will see `HTML` and `Video` xblock created.

**Author notes and concerns**:

1. This PR doesn't handle if the links are not in csv file.
1. We need a valid Edx Video Id with course Id to test the Edx Id approach but we can easily test youtube Id approach, since this is working the Edx Id should work as well.

**Reviewers**
